libc/startup/baremetal/arm/start.cpp

```diff
@@ -131,30 +131,40 @@ namespace LIBC_NAMESPACE_DECL {
   __arm_wsr("CPSR_c", 0x13); // SVC
 #endif
 
-#ifdef __ARM_FP
-// Enable FPU
-#if __ARM_ARCH_PROFILE == 'M'
+#if __ARM_ARCH_PROFILE == 'M' &&                                               \
+    (defined(__ARM_FP) || defined(__ARM_FEATURE_MVE))
+  // Enable FPU and MVE. They can't be enabled independently: the two are
+  // governed by the same bits in CPACR.
   // Based on
   // https://developer.arm.com/documentation/dui0646/c/Cortex-M7-Peripherals/Floating-Point-Unit/Enabling-the-FPU
-  // Set CPACR cp10 and cp11
-  auto cpacr = (volatile uint32_t *const)0xE000ED88;
+  // Set CPACR cp10 and cp11.
+  auto cpacr = reinterpret_cast<volatile uint32_t *const>(0xE000ED88);
   *cpacr |= (0xF << 20);
   __dsb(0xF);
   __isb(0xF);
-#elif __ARM_ARCH_PROFILE == 'A' || __ARM_ARCH_PROFILE == 'R'
+#if defined(__ARM_FEATURE_MVE)
+  // Set FPSCR's LTPSIZE field to 4 to disable low-overhead-loop tail
```

Comment:
Sorry to nitpick, but this comment could be misleading. "Disabling" tail predication sounds like making it impossible for people to use it, and we're not doing that. We're only ensuring that it doesn't accidentally trigger at startup.

Perhaps an alternative wording along the lines of "initialize low-overhead-loop tail predication to its neutral state"?

Reply:
Thank you. I used your suggestion verbatim as I can't come up with something better anyway